python/src/scipp/plot/plot3d.py

```diff
@@ -78,6 +72,15 @@ def __init__(self,
                                  dim_label_map=self.dim_label_map,
                                  positions=positions)
 
+        # Create control widgets (sliders and buttons)
+        self.widgets = PlotWidgets(axes=self.axes,
+                                   ndim=view_ndims,
+                                   name=self.name,
+                                   dim_to_shape=self.dim_to_shape,
+                                   masks=self.masks,
+                                   positions=positions,
+                                   multid_coord=self.model.get_multid_coord())
```

Comment:
Almost identical code 3 times, is another common base class possible? Or maybe most of these arguments should be stored in a class or dict, such that the duplicate code is less, something like `self.widgets = PlotWidget(self.plot_params)`?

Not suggesting that this must be addressed in the PR, just something to think about for the next round of cleanup.

Reply:
Yes, I think I prefer the dict option. I suggest delaying to a later cleanup as not relevant to this PR